CONTRIBUTING.md

```diff
@@ -1,13 +1,16 @@
 # Contributing to Rich
 
-This project welcomes contributions in the form of Pull Requests. For clear bug-fixes / typos etc. just submit a PR. For new features or if there is any doubt in how to fix a bug, you might want to open an issue prior to starting work, or email willmcgugan+rich@gmail.com to discuss it first.
+This project welcomes contributions in the form of Pull Requests. 
+For clear bug-fixes / typos etc. just submit a PR. 
+For new features or if there is any doubt in how to fix a bug, you might want 
+to open an issue prior to starting work, or email willmcgugan+rich@gmail.com 
+to discuss it first.
 
 ## Development Environment
 
-To start developing with Rich, first create a _virtual environment_ then run the following to install development requirements:
+To start developing with Rich, run the following to install requirements:
 
 ```
-pip install -r requirements-dev.txt
```

Comment:
Fair point, it won't really. I guess they would just need to pip install poetry instead of the requirements-file. My impression was that you would need poetry to install the package dependencies anyways so why not unify the two sets of dependencies.

Feel free to close this if you prefer to have them both; if not I could tweak the contributing doc a little

Reply:
Just a note that [poetry doesn't recommend installing via pip directly](https://python-poetry.org/docs/#installing-with-pip).

Reply:
> Just a note that poetry doesn't recommend installing via pip directly.

True, but for the purpose of contributing to a repo, I think it might be a big ask to tell people to curl poetry :p The only feature I know of that you miss out on from the pip install  is `poetry self update` which isn't needed in this context

Reply:
Might be useful to list both options though

Reply:
If we're going to go the poetry route, I think all we need to do is document you need to install poetry and link to https://python-poetry.org/docs/#installing-with-pip

Is it much of a hassle to install with curl? It is a little surprising certainly. But I'd hope that a dev motivated enough to consider contributing would stick with it.

Reply:
GitHub codespaces will also come in handy for making it easy to setup development environments once it is fully released.

Reply:
Updated to

```md
Rich uses [poetry](https://python-poetry.org/docs/) for packaging and 
dependency management. To start developing with Rich, either install the 
recommended way (using curl) as described in their 
[docs](https://python-poetry.org/docs/#installation), or simply run:

`pip install poetry`

Once Poetry is installed, run

`poetry install`

To install dependencies
```